### DISCUSSION TITLE
automating mobile Safari

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#15953] [Q&A] [answered]
I'm trying to automate some tests for a website with the safari browser on an simulator.
I use Appium 2.0 latest beta under Big Sur.
But i the following  error on locating elements

`Unsupported locator strategy: -ios class chain`

**driver setup:**
`driver = new IOSDriver<IOSElement>(new URL("http://127.0.0.1:4727/wd/hub"), caps);`

**Caps:**
`   "platformName": "ios",
     "appium:app": "",
    "appium:automationName": "XCUITest",
    "appium:browsername": "Safari",
    "appium:bundleId": "com.apple.mobilesafari",
    "appium:connectHardwareKeyboard": false,
    "appium:deviceName": "iPhone 8 Plus",
    "appium:keepKeyChains": false,
    "appium:language": "de",
    "appium:launchTimeout": 60000,
    "appium:locale": "de_DE",
    "appium:newCommandTimeout": 120,
    "appium:platformVersion": "14.5",
    "appium:sendKeyStrategy": "setValue",
    "appium:udid": "E00B2CE0-6F32-4C11-BD3F-B5A1E6D88B24",
    "appium:useNewWDA": true,
    "appium:waitForIdleTimeout": 1,
    "appium:wdaLocalPort": 8301,
    "appium:noReset": "false"`
    
[full_log.log](https://github.com/appium/appium/files/7327856/full_log.log)

**Comment by jlipps** [comment, marked as answer]
the `-ios class chain` strategy is for native contexts, not web contexts. With web contexts you're limited to the standard webdriver locator strategies.

**Reply by RSM-SBA** [reply]
Thank your for the quick response
